plugin.program.autowidget/resources/lib/common/utils.py

```diff
@@ -427,7 +427,7 @@ def remove_file(file):
 def read_file(file):
     content = None
     if xbmcvfs.exists(file):
-        with xbmcvfs.File(file) as f:
+        with contextlib.closing(xbmcvfs.File(file)) as f:
```

Comment:
I don't think it causes an issue. I didn't think it was needed but I'm having a lot of problems with the .widget files being corrupted and not being able to be read. Now that I remove them when that happens it results in the static widget asking again which path to use which is very annoying since it doesnt tell me which widget its for.
I haven't got to the bottom of why they are getting corrupted. It should not be happening this often. kodi does crash often on my firestick and also its process manager kills it in the background when switching to something else.

Reply:
Yeah, we should probably try not to do that 😁